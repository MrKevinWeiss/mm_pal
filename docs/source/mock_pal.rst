mock\_pal package
=================

Submodules
----------

mock\_pal.mock\_cli module
--------------------------

.. automodule:: mock_pal.mock_cli
   :members:
   :undoc-members:
   :show-inheritance:

mock\_pal.mock\_dev module
--------------------------

.. automodule:: mock_pal.mock_dev
   :members:
   :undoc-members:
   :show-inheritance:

mock\_pal.mock\_if module
-------------------------

.. automodule:: mock_pal.mock_if
   :members:
   :undoc-members:
   :show-inheritance:


Module contents
---------------

.. automodule:: mock_pal
   :members:
   :undoc-members:
   :show-inheritance:
