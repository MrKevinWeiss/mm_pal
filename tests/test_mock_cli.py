import pytest
from conftest import MM_PATH, EXT_PORT
from mock_pal import MockCli


@pytest.mark.parametrize("kwargs", [{},
                                    {'mm_path': MM_PATH},
                                    {'mem_map': []},
                                    {'driver_type': 'serial'},
                                    {"parser_type": 'json'}])
def test_init(mock_app_json, capsys, kwargs):
    tmp = MockCli(port=EXT_PORT, **kwargs).onecmd("get_version")
    cap = capsys.readouterr()
    assert 'Interface version: 0.0.1' in cap.out


@pytest.mark.parametrize("cmd", ["read_reg i8",
                                 "write_reg i8 1",
                                 "read_struct stt",
                                 "commit",
                                 "soft_reset",
                                 "get_version",
                                 "info_reg arru8",
                                 "info_reg",
                                 "info_param name",
                                 "info_param nothing",
                                 "special_cmd",
                                 "key_error"])
def test_commands(mock_app_json, capsys, regtest, cmd):
    tmp = MockCli(port=EXT_PORT, persistent_history_file="/tmp").onecmd(cmd)
    cap = capsys.readouterr()
    regtest.write(cap.out)


def test_completion(mock_app_json):
    cli = MockCli(port=EXT_PORT)
    assert "arru8" in cli.regs_choices_method()
    assert "name" in cli.param_choices_method()
