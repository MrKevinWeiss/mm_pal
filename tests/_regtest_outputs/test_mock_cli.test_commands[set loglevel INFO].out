loglevel - was: 'WARNING'
now: 'INFO'
